mlflow/system_metrics/metrics/base_metrics_monitor.py

```diff
@@ -0,0 +1,34 @@
+"""Base class of system metrics monitor."""
+import abc
+from collections import defaultdict
+
+
+class BaseMetricsMonitor(abc.ABC):
+    """Base class of system metrics monitor.
+
+    Args:
+        name: string, name of the monitor.
+    """
+
+    def __init__(self, name):
+        self.name = name
+
+        self._metrics = defaultdict(list)
+
+    @abc.abstractmethod
+    def collect_metrics(self):
+        raise NotImplementedError
```

Comment:
Can we add a docstring here and describe what should be implemented, and remove `raise NotImplementedError`?